tests/test_iib_client.py

```diff
@@ -298,10 +337,66 @@ def test_iib_client_no_overwrite_from_index_or_token(
                 cnr_token="cnr",
                 organization="org",
                 overwrite_from_index_token="str",
+                deprecation_list=None,
+            )
+
+
+def test_iib_client_deprecation_list(
+    fixture_add_build_details_json,
+    fixture_deprecation_list_build_details_json,
+):
+    with requests_mock.Mocker() as m:
```

Comment:
I see this piece has been duplicated at a bunch of places. Can we please put this into a fixture and use it?

Reply:
It is the same approach but the request is not the same. It is used for api calls but here there are for `add` later on there are for `rm` and `regenerate-bundle`. The calls are specific to models.

Reply:
@lipoja any ideas?

Reply:
Right now from top of my head no - I agree with @pbortlov that those are different cases - different `register_uri` parameters. Can not be changed into one fixture.

Reply:
@yashvardhannanavati is it fine to resolve this comment?

Reply:
Maybe we can create separate fixtures for mocking add, rm ... so it can be shared among multiple test cases. But I think  it might not be worth it.

Reply:
@lipoja that's what I meant. Creating fixtures fpr add, rm etc. For starters, since we are only dealing with the `add` request here, let's change that to a fixture? 

If both of you feel strongly against this suggestion then I'm willing to let it go. But these were just my two cents for better coding practices :)

Reply:
For this test case we have `add` here but if you check the whole code we have also `rm`, `builds`, ... I think almost everything. 
So we should create fixtures for all of those requests - or ideally one excepting parameters, but then you are basically back to calling this code.

Let's see what @pbortlov thinks about it. Maybe we can think about creating fixtures of every endpoint that we have? But I am not sure if it is that helpful. (except of setting URL path at one place) because other parameters should be allowed to change.

Reply:
@yashvardhannanavati @lipoja I do have a feeling this should be a part of code refactoring not this PR. As Jan mentioned it is not only `add` request. If we change it here we would have to change it everywhere (`rm`, `builds`) and it would not save code lines. 
The mocker is consists of 2 parameters that have to be changed "uri" part `/api/v1/builds` and JSON which is not the same.

If we really want to change it we should create a refactoring story.

Reply:
@pbortlov Okay, could you please file a backlog story for it? :)

Reply:
https://projects.engineering.redhat.com/browse/CLOUDDST-5218